sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -187,6 +187,64 @@ case class Like(left: Expression, right: Expression)
   override def matches(regex: Pattern, str: String): Boolean = regex.matcher(str).matches()
 
   override def toString: String = s"$left LIKE $right"
+
+  override protected def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
+    val patternClass = classOf[Pattern].getName
+
+    val literalRight: String = right match {
+      case x @ Literal(value: String, StringType) => escape(value)
+      case _ => null
+    }
+
+    val leftGen = left.gen(ctx)
+    val rightGen = right.gen(ctx)
+
+    val patternCode =
+      if (literalRight != null) {
+        s"${patternClass} pattern = $patternClass.compile($literalRight);"
+      } else {
+        s"""
+          StringBuilder regex = new StringBuilder("(?s)");
```

Comment:
I am not sure, if `StringBuilder`is imported. If not define somewhere `val sb = classOf[StringBuilder].getName` and use `$sb`

You shouldn't use `regex`. You can create a save variable name with `ctx.freshName`